### PR TITLE
Support non-interactive use of ensureVenv script

### PR DESCRIPTION
**Pull request by LeonarddeR**
### Link to issue number:
https://github.com/nvaccess/nvda/pull/13048#issuecomment-1178807953

### Summary of the issue:
#13048 introduced another prompt to adres when dependencies in requirements.txt have changed. This blocks non-interactive use of the ensureVenv script, e.g. when running "scons source" non-interactively.

### Description of user facing changes
As proposed by @feerrenrut in https://github.com/nvaccess/nvda/pull/13048#issuecomment-1178807953, just use defaults when running in non-interactive mode.

### Description of development approach
See https://github.com/nvaccess/nvda/pull/13048#issuecomment-1178807953. Note that isatty is not always avaliable on sys.stdout, e.g. when it is set to the NVDA python console.

### Testing strategy:
Tested running `scons source > d:\output` and observed that the venv was dropped and recreated.

### Known issues with pull request:
Do we want to present the question and also print that the question has been answered as default due to running in non-interactive, I figured it would be a bit overkill for that purpose.

### Change log entries:
None needed

### Code Review Checklist:

- [x] Pull Request description:
  - description is up to date
  - change log entries
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] API is compatible with existing add-ons.
- [x] Documentation:
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English

### CODE REVIEW

File: venvUtils/ensureVenv.py [hunk 1]
```diff
@@ -19,15 +19,20 @@
...
 venv_orig_requirements_path: str = os.path.join(venv_path, "_requirements.txt")
 venv_python_version_path: str = os.path.join(venv_path, "python_version")
+#: Whether this script is run interactively,
+#: i.e. whether user input is possible to answer questions.
+#: Value is True if interactive (i.e. stdout is attached to a terminal), False otherwise.
+isInteractive = hasattr(sys.stdout, "isatty") and sys.stdout.isatty()
 
 
-def askYesNoQuestion(message: str) -> bool:
+def askYesNoQuestion(message: str, default: bool = True) -> bool:
```
Comment (feerrenrut) [inline venvUtils/ensureVenv.py:28]: I think the default should be explicit. There are only two usages.

**Comment by feerrenrut** [comment]
> Do we want to present the question and also print that the question has been answered as default due to running in non-interactive, I figured it would be a bit overkill for that purpose.

I think it would be a good idea to have a record of this in the build output. If this does the wrong thing in some situation it will be much easier to track it down.

File: venvUtils/ensureVenv.py [hunk 2]
```diff
@@ -25,24 +25,26 @@
...
 
-def askYesNoQuestion(message: str, default: bool = True) -> bool:
+def askYesNoQuestion(message: str) -> bool:
 	"""
 	Displays the given message to the user and accepts y or n as input.
 	Any other input causes the question to be asked again.
-	If isInteractive is False, the default is always returned.
+	If isInteractive is False, y is assumed and True is returned.
 	@returns: True for y and False for n.
 	"""
```
Comment (feerrenrut) [inline venvUtils/ensureVenv.py:34]: This is still implicit.

Instead:
```python
def askYesNoQuestion(message: str, default: bool) -> bool:
	"""
	Displays the given message to the user and accepts y or n as input.
	Any other input causes the question to be asked again.
	If isInteractive is False, the default is always returned, the question and outcome will still be sent to stdout for inspection of the build.
	@param default: the return value when the user can not be prompted.
	@returns: True for y and False for n.
	"""
```

Comment (leonardder) [reply venvUtils/ensureVenv.py:34]: Ah I interpreted explicit the other way around 🤣

**Review by feerrenrut — APPROVED** [review]
Thanks for your work here @leonardder

**Pull request merged**